### PR TITLE
Update `AstroInstance` interface reference

### PR DESCRIPTION
**Pull request by MoustaphaDev**
#### What kind of changes does this PR include?

- New or updated content

#### Description

The update the [`AstroInstance`](https://github.com/withastro/astro/blob/main/packages/astro/src/@types/astro.ts#L983-L986) interface

### CODE REVIEW

File: src/pages/en/reference/api-reference.md [hunk 1]
```diff
@@ -86,6 +86,8 @@ Astro files have the following interface:
 
 ```ts
 export interface AstroInstance {
+  file: string;
```
Comment (delucis) [inline src/pages/en/reference/api-reference.md:89]: Do we want to add some comments like we do for `MarkdownInstance`?

```suggestion
  /* The file path of this file */
  file: string;
```

File: src/pages/en/reference/api-reference.md [hunk 2]
```diff
@@ -86,6 +86,8 @@ Astro files have the following interface:
...
+	url: string | undefined;
```
Comment (delucis) [inline src/pages/en/reference/api-reference.md:90]: Maybe we should also hint at _why_ this is sometimes `undefined`?

```suggestion
  /* The URL for this file (if it is in the pages directory) */
	url: string | undefined;
```

**Review by delucis — COMMENTED** [review]
Thanks @MoustaphaDev, this is helpful! Left some comments.

File: src/pages/en/reference/api-reference.md (hunk 1 above)
Comment (MoustaphaDev) [reply src/pages/en/reference/api-reference.md:90]: Thanks for the suggestion!

**Review by delucis — APPROVED** [review]
Thanks! Great improvement 🙌

**Pull request merged**